File: Arrays/LongestSubArray.java
package Arrays;

public class LongestSubArray {
    public static int longestSubArray(int[] nums, int total){
        int n = nums.length;
        int maxlen = 0;
        for (int i = 0; i < n; i++){
            for(int j = 0; j < n; j ++){
                long sum = 0;
                for(int k = i; k <= j; k++){
                    sum += nums[k];
                }
                if(sum == total){
                    maxlen = Math.max(maxlen, j - i + 1);
                }
            }
        }
        return maxlen;
    }

    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 4, 5};
        int total = 5;
        int result = longestSubArray(nums, total);
        System.out.println("Longest Subarray with sum " + total + ": " + result);
    }
}

File: Arrays/LongestSubArray.py
def LongestSubArray(arr, total):
    n = len(arr)
    maxlen = 0
    for i in range(0, n ):
        for j in range(0, n):
            sum = 0
            for k in range(i, j + 1):
                sum += arr[k]
            
            if sum == total:
                maxlen = max(maxlen, j - i + 1)
                
    return maxlen

# Example usage:
arr = [1, 2, 3, 4, 5]
total = 9
result = LongestSubArray(arr, total)
print(f"The length of the longest subarray with sum {total} is: {result}")
